### PR TITLE
Always include ANCM in build output

### PR DESCRIPTION
**Pull request by jkotalik**
Note, this is purely a build only change. I'm currently working on verifying this works with separate branches on aspnetci.

## Impact
After we stopped pushing to MyGet, ANCM transfer packages have stopped being uploaded. Builds are now failing with an error saying ANCM could not be found as an artifact dependency as well as the latest version of ANCM being 2.2.2 on myget. This change makes it so ANCM always builds, regardless of if there was a change to ANCM or not.

The reason this is fine is:
- ANCM is shipped globally and isn't put in a package.
- We are already doing this in 3.0.

## Risk
Very low risk. We are rebuilding ANCM every patch version, which will update dll/msi versions. However, the source code being used is the same.

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <!-- Always include framework metapackages in patch updates. -->
     <IsPackageInThisPatch Condition="'$(IsFrameworkMetapackage)' == 'true' OR '$(IsSharedSourcePackage)' == 'true' ">true</IsPackageInThisPatch>
     <IsPackageInThisPatch Condition="'$(IsPackageInThisPatch)' == ''">$(PackagesInPatch.Contains(' $(PackageId);'))</IsPackageInThisPatch>
+    <IsPackageInThisPatch Condition="'$(IsANCMPackage)' == 'true'">true</IsPackageInThisPatch>
```
Comment (JunTaoLuo) [inline Directory.Build.targets]: Add a comment explaining why this must be rebuilt for every patch.

**Review by JunTaoLuo — APPROVED** [review]
Yes this is a fix that we need in order to build our 2.2 patch. @anurse what kind of approval process is needed for this change?

**Comment by anurse** [comment]
> Yes this is a fix that we need in order to build our 2.2 patch.

👍 . Make sure `netcoreship` is **aware** of the change, but if it's necessary to build, we gotta take it so go ahead.

**Comment by jkotalik** [comment]
Verified here: http://aspnetci/viewLog.html?buildId=655457&buildTypeId=Releases_22xPublic_WindowsInstallers&tab=buildLog&_focus=481#_state=441

**Comment by vivmishra** [comment]
Approved for 2.2.6

**Comment by JunTaoLuo** [comment]
I'm going to merge this one since we are running into build hangs on 2.2 right now and it's unlikely that the tests will pass. The linux failure is unrelated.

**Pull request merged**